src/compiler/ast.h

```diff
@@ -269,14 +269,13 @@ namespace verona::compiler
 
   struct Member : public SourceLocatableMixin<ASTContainer>
   {
-    ASTChild<Name> name;
-
     /**
      * Pointer back to the Entity which contains this member.
      * Added during resolution
      */
     const Entity* parent = nullptr;
 
+    virtual const Name& get_name() const = 0;
```

Comment:
Why do you want this to be virtual? And why did you drop the common field from Member.  Is this due to Pegmatite restrictions on parsing in order of fields?

Reply:
Pegmatite requires fields to be laid out in the order they appear in the syntax, where fields of the base type come before fields of the derived class.

Previously both Field and Method started with their name, so it was put in Entity. But because Method now has a kind before name, we need to move the name into the derived class.

We want this virtual method so we can get the name of a Entity*.